misc/output_test.py

```diff
@@ -152,14 +153,18 @@ class Output(unittest.TestCase):
         '',
     ))
 
-    def _test_expected_error(self, plan: str, flags: T.Optional[str], expected: str):
+    def _test_expected_error(self,  expected: str, *args, **kwargs):
```

Comment:
Please avoid changing the order of arguments if possible, as this reduces the number of potential merge / rebase conflicts for all other unsubmitted PRs. In this specific case, I believe it should be possible to add the extra args at the end of the current signature without requiring changing other call sites.

Reply:
About keeping as few changes in signatures as possible to reduce conflicts, this specific case limits the usage of calling the following `run` function.

Passing every possible argument into `run` in the wrong order, as I've done with [env](https://github.com/ninja-build/ninja/pull/2540/files#diff-32108e2fd2aaf591ed4fae7d0955d688432825bc286bdd99a4962a7db3a8d6a2R317) would be a mess. 

Keeping the current order would mean something like `(self, plan: str, flags: T.Optional[str], expected: str, exit_code: Optional[int] = 0, env: Dict[str, str] = {})` or `(self, plan: str, flags: T.Optional[str], expected: str, env: Dict[str, str] = {}, exit_code: Optional[int] = 0)`, and it means we split run-related args with the method-related. IMHO, it doesn't look nice, and adding more run-related args in the future will be even more problematic.

Is it worth changing it now until the method is not spread widely?

Reply:
I think there is no point in making all arguments explicit, i.e. you should be able to just add `*args, **kwargs` at the end of the current definition, and check `kwargs["exit_code"]` manually.

Reply:
The issue, `flags: T.Optional[str],` doesn't have the default value, although it has it in both [run](https://github.com/ninja-build/ninja/blob/d3444792771cfef3d47174e3535bab63e4c97197/misc/output_test.py#L80) and [run](https://github.com/ninja-build/ninja/blob/d3444792771cfef3d47174e3535bab63e4c97197/misc/output_test.py#L133).

It is worth having a default for `None`, but the following `expected` without defaults must stay in front of `flags` then.

To be honest, the current signature is very questionable. If it's a blocker, I could leave the first arguments as they are and pass all the `**kwargs`.

Reply:
```diff
diff --git a/misc/output_test.py b/misc/output_test.py
index 641a444..b3e2a15 100755
--- a/misc/output_test.py
+++ b/misc/output_test.py
@@ -153,7 +153,7 @@ class Output(unittest.TestCase):
         '',
     ))
 
-    def _test_expected_error(self,  expected: str, *args, **kwargs):
+    def _test_expected_error(self, plan: str, flags: T.Optional[str], expected: str, *args, **kwargs):
         """Run Ninja with a given plan and flags, and verify its cooked output against an expected content.
         All *args and **kwargs are passed to the `run`, except `exit_code`
         """
@@ -161,7 +161,7 @@ class Output(unittest.TestCase):
         kwargs['print_err_output'] = False
         exit_code = kwargs.pop("exit_code", None)
         with self.assertRaises(subprocess.CalledProcessError) as cm:
-            run(*args,  **kwargs)
+            run(plan, flags, *args,  **kwargs)
         actual = cm.exception.cooked_output
         if exit_code is not None:
             self.assertEqual(cm.exception.returncode, exit_code)
@@ -311,42 +311,48 @@ build success: CUSTOM_COMMAND
         env = default_env.copy()
         env['TERM'] = 'dumb'
         self._test_expected_error(
+            plan, '124',
             f'''[1/1] {py} -c 'exit(124)'
 FAILED: [code=124] 124 \n{py} -c 'exit(124)'
 ninja: build stopped: subcommand failed.
 ''',
-            plan, '124', exit_code=124, env=env,
+            exit_code=124, env=env,
         )
         self._test_expected_error(
+            plan, '127',
             f'''[1/1] {py} -c 'exit(127)'
 FAILED: [code=127] 127 \n{py} -c 'exit(127)'
 ninja: build stopped: subcommand failed.
 ''',
-            plan, '127', exit_code=127, env=env,
+            exit_code=127, env=env,
         )
         self._test_expected_error(
+            plan, '130',
             'ninja: build stopped: interrupted by user.\n',
-            plan, '130', exit_code=130, env=env,
+            exit_code=130, env=env,
         )
         self._test_expected_error(
+            plan, '137',
             f'''[1/1] {py} -c 'exit(137)'
 FAILED: [code=137] 137 \n{py} -c 'exit(137)'
 ninja: build stopped: subcommand failed.
 ''',
-            plan, '137', exit_code=137, env=env,
+            exit_code=137, env=env,
         )
         self._test_expected_error(
+            plan, 'non-existent-target',
             "ninja: error: unknown target 'non-existent-target'\n",
-            plan, 'non-existent-target', exit_code=1, env=env,
+            exit_code=1, env=env,
         )
         self._test_expected_error(
+            plan, '-j2 success 127',
             f'''[1/2] {py} -c 'exit(127)'
 FAILED: [code=127] 127 \n{py} -c 'exit(127)'
 [2/2] sleep 0.3; echo success
 success
 ninja: build stopped: subcommand failed.
 ''',
-            plan, '-j2 success 127', exit_code=127, env=env,
+            exit_code=127, env=env,
         )
 
 
@@ -478,6 +484,7 @@ build out4 : cat in4
 
 
         self._test_expected_error(
+            plan, '-t compdb-targets',
             '''ninja: error: compdb-targets expects the name of at least one target
 usage: ninja -t compdb [-hx] target [targets]
 
@@ -485,17 +492,16 @@ options:
   -h     display this help message
   -x     expand @rspfile style response file invocations
 ''',
-            plan, '-t compdb-targets',
         )
 
         self._test_expected_error(
-            "ninja: fatal: 'in1' is not a target (i.e. it is not an output of any `build` statement)\n",
             plan, '-t compdb-targets in1',
+            "ninja: fatal: 'in1' is not a target (i.e. it is not an output of any `build` statement)\n",
         )
 
         self._test_expected_error(
-            "ninja: fatal: unknown target 'nonexistent_target'\n",
             plan, '-t compdb-targets nonexistent_target',
+            "ninja: fatal: unknown target 'nonexistent_target'\n",
         )
```

Like this?

Reply:
the patch is applied in the latest rebase